src/frontend/src/tests/lib/stores/convert.store.spec.ts

```diff
@@ -0,0 +1,34 @@
+import { ETHEREUM_TOKEN, ICP_TOKEN } from '$env/tokens.env';
+import { initConvertContext } from '$lib/stores/convert.store';
+import { mockPage } from '$tests/mocks/page.store.mock';
+import { testDerivedUpdates } from '$tests/utils/derived.utils';
+import { expect } from 'vitest';
+
+describe('convertStore', () => {
+	beforeEach(() => {
+		mockPage.reset();
+	});
+
+	it('should ensure derived stores update at most once when the store changes', async () => {
+		await testDerivedUpdates(() =>
+			initConvertContext({
+				destinationToken: ETHEREUM_TOKEN,
+				sourceToken: ICP_TOKEN
+			})
+		);
+	});
+
+	it('should have all expected properties', () => {
```

Comment:
Can you extend the test one step further be feeding some data in the `balanceStore` and `exchanges` and assert that the derived values are expected?

Reply:
Yes, sure, should have done it right away 🙂